doc/adr/0016- end-to-end testing-strategy.rst

```diff
@@ -0,0 +1,199 @@
+.. _end_to_end_testing_strategy:
+
+ADR 16: End-to-end testing strategy for plutus, cardano-ledger-api and cardano-node-client
+==========================================================================================
+
+Date: 2022-12-02
+
+Author(s)
+---------
+
+james-iohk <james.browning@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+End-to-end testing of Plutus functionality is curently performed by a combination of
+automation and exploratory approaches. Both are performed on a Cardano testnet with a real node.
```

Comment:
Maybe replace the word "testnet" with "preview" and "preprod" networks in this document?